lib/components/home/artwork_rail.js

```diff
@@ -54,12 +56,19 @@ class ArtworkRail extends React.Component {
     // pop new view controller
   }
 
+  onGridLayout(event) {
+    console.log('height: ' + event.nativeEvent.layout.height);
+    this.setState({ gridHeight: event.nativeEvent.layout.height });
+  }
+
   renderModuleResults() {
     if (this.props.rail.results && this.props.rail.results.length) {
       return (
-        <View style={[styles.container, { height: this.state.expanded ? 2350 : 500 }]}>
-          <View style={[styles.gridContainer, { height: this.state.expanded ? 2300 : 420 }]}>
-            <Grid artworks={this.props.rail.results}/>
+        <View style={[styles.container, { height: this.state.expanded ? this.state.gridHeight + 150 : 500 }]}>
+          <View style={[styles.gridContainer, { height: this.state.expanded ? this.state.gridHeight + 100 : 420 }]}>
+            <View onLayout={this.onGridLayout.bind(this)}>
```

Comment:
[I have](https://github.com/artsy/emission/pull/220/files#diff-2f5aa3c37ef9f2653a0096c9f8344357L30), maybe the context is different though - it's early days for me, can you try `<View onLayout={this.onGridLayout}>`

If that doesn't work - merge away, and I'll note to check out the language semantics later